e2e/tests/node_reboot.robot

```diff
@@ -129,26 +155,64 @@ Power Off All Worker Nodes For More Than Pod Eviction Timeout While Workload Hea
     END
 
 Reboot Volume Node While Workload Heavy Writing
-    Given Create statefulset 0 with ${VOLUME_TYPE} volume
+    Given Create statefulset 0 using ${VOLUME_TYPE} volume
     FOR    ${i}    IN RANGE    ${LOOP_COUNT}
-        And Keep writing data to statefulset 0
+        And Keep writing data to pod of statefulset 0
 
         When Reboot volume node of statefulset 0
         And Wait for volume of statefulset 0 healthy
-        And Wait for statefulset 0 stable
+        And Wait for statefulset 0 pods stable
 
         Then Check statefulset 0 works
     END
 
 Power Off Volume Node For More Than Pod Eviction Timeout While Workload Heavy Writing
-    Given Create statefulset 0 with rwo volume
+    Given Create statefulset 0 using RWO volume
 
     FOR    ${i}    IN RANGE    ${LOOP_COUNT}
-        And Keep writing data to statefulset 0
+        And Keep writing data to pod of statefulset 0
 
-        When Power off volume node of statefulset 0 for 6 mins
+        When Power off volume node of statefulset 0 for 6 minutes
         And Wait for volume of statefulset 0 healthy
-        And Wait for statefulset 0 stable
+        And Wait for statefulset 0 pods stable
 
         Then Check statefulset 0 works
     END
+
+Reboot Volume Node While Heavy Writing And Recurring Jobs Exist
```

Comment:
This test case failed with message:
```
No new snapshot created by recurringjob e2e-test-volume-1-snap for e2e-test-volume-1 since 2024-03-05 13:04:43.028887
```
Could you help check it? Thank you!
https://ci.longhorn.io/job/private/job/longhorn-e2e-test/375/robot/Tests/Node%20Reboot/Reboot%20Volume%20Node%20While%20Heavy%20Writing%20And%20Recurring%20Jobs%20Exist/

Reply:
Here is my test result:
```
Reboot Volume Node While Heavy Writing And Recurring Jobs Exist       | PASS |
------------------------------------------------------------------------------
Tests.Node Reboot :: Negative Test Cases                              | PASS |
1 test, 1 passed, 0 failed
==============================================================================
Tests                                                                 | PASS |
1 test, 1 passed, 0 failed
==============================================================================
```

I've tried testing a couple of times, but it might not be enough to hit the failure. Based on the result and the log, this could be due to either flaky or a genuine issue, however, most likely unrelated to this PR. If you encounter this error again, please open a separate issue for tracking. This us to continue reviewing this PR. Thank you.

Reply:
Since I can find changes in `recurringjob/rest.py`, if they're not related to this PR, it's better to keep it intact to make things clear.

Reply:
I will take another look.

Reply:
It should be fixed now. I've run the test 10 times and cannot reproduce the error. PTAL, thank you.

Reply:
I've noticed the test is failing https://ci.longhorn.io/job/private/job/longhorn-e2e-test/391/, and based on the support bundle, it seems the backup target wasn't set.

So I tried to test with the same scenario and observed nearly a 10-minute gap between each of the backups:
```shell
> k -n longhorn-system get backup -l backup-volume=e2e-test-volume-0
NAME                      SNAPSHOTNAME   SNAPSHOTSIZE   SNAPSHOTCREATEDAT      STATE   LASTSYNCEDAT
backup-f5856f98433441c7                                 2024-03-07T23:47:28Z   Error   2024-03-07T23:47:33Z
backup-ab066cb214904499                                 2024-03-07T23:39:50Z   Error   2024-03-07T23:39:56Z
backup-8d88e90de24c44e1                                 2024-03-08T00:00:47Z   Error   2024-03-08T00:00:55Z
backup-e0720ab2a3154c06                                 2024-03-08T00:08:26Z   Error   2024-03-08T00:08:31Z
backup-fa4f38a718fb4e3d                                 2024-03-08T00:14:21Z   Error   2024-03-08T00:14:31Z
backup-6caabe3a17af49a0                                 2024-03-08T00:23:10Z   Error   2024-03-08T00:23:15Z
backup-7fa65faafe354c6e                                 2024-03-08T00:29:37Z   Error   2024-03-08T00:29:45Z
```
Therefore, I've updated the PR to wait for 10 minutes instead of 2 minutes.

Reply:
It's a known issue. That's why https://github.com/longhorn/longhorn-tests/pull/1539 has been required to be merged long time ago.

Reply:
Got it. Since the backup will still get created with some delay. So increasing the wait time should enable the test case to pass, as evidence that the recurring job is working correctly. Or do you think we should maintain the wait time at 2 minutes?

Reply:
If you run it repeatedly for long enough it could still flaky, I've made modification in https://github.com/longhorn/longhorn-tests/pull/1539.

Reply:
Found another kind of error:
```
ApiException: (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Audit-Id': '2d759e7c-f157-40f1-b7f4-e60a3ce74b03', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Kubernetes-Pf-Flowschema-Uid': '5501a88c-4f74-457c-82ee-79c16bf9ea40', 'X-Kubernetes-Pf-Prioritylevel-Uid': 'e665ab9b-e2f2-4eff-84ef-e74da9e5f588', 'Date': 'Fri, 08 Mar 2024 05:51:14 GMT', 'Content-Length': '194'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \"ip-10-0-2-123\" not found","reason":"NotFound","details":{"name":"ip-10-0-2-123","kind":"pods"},"code":404}



Also teardown failed:
ApiException: (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Audit-Id': 'e4143121-7fb8-4f30-8392-13f2f9e4a353', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Kubernetes-Pf-Flowschema-Uid': '5501a88c-4f74-457c-82ee-79c16bf9ea40', 'X-Kubernetes-Pf-Prioritylevel-Uid': 'e665ab9b-e2f2-4eff-84ef-e74da9e5f588', 'Date': 'Fri, 08 Mar 2024 05:51:14 GMT', 'Content-Length': '194'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \"ip-10-0-2-123\" not found","reason":"NotFound","details":{"name":"ip-10-0-2-123","kind":"pods"},"code":404}
```
https://ci.longhorn.io/job/private/job/longhorn-e2e-test/408/robot/Tests/Node%20Reboot/Reboot%20Volume%20Node%20While%20Heavy%20Writing%20And%20Recurring%20Jobs%20Exist/

What has changed in this PR today?

Reply:
> What has changed in this PR today?

- Fixed the casings for keyword: https://github.com/longhorn/longhorn-tests/compare/34813320e3576ede3ff12a1246d987d1504f2c61..74d1360e984188410b4c52df68be33269dc56938
- Increased the iteration time: https://github.com/longhorn/longhorn-tests/compare/773ed11c1f437943aafd528e6f7e77e04de1fefb..f9d34c75cfe31cc9f753cbe6aa46c9c83c41133a
- Rebases.

```
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \"ip-10-0-2-123\" not found","reason":"NotFound","details":{"name":"ip-10-0-2-123","kind":"pods"},"code":404}
```
Weird, seems like its looking for a pod with a node name.

Reply:
![Screenshot from 2024-03-08 14-46-37](https://github.com/longhorn/longhorn-tests/assets/22583541/9cd34f5e-1444-4e37-b058-8a05a0edc533)
I couldn't reproduce this locally. And it shouldn't be looking for a pod there.

![Screenshot from 2024-03-08 14-48-38](https://github.com/longhorn/longhorn-tests/assets/22583541/9ba3735f-de66-4336-a11f-b270058c07d2)

As discussed, please start another test with the TRACE log so we can see why it's doing that.

Reply:
Spot this could be happening in the [launch_pod](https://github.com/longhorn/longhorn-tests/blob/a868979cb121ab3c6098ce4fda1c7698e494643a/e2e/libs/node_exec/node_exec.py#L95-L98) during [write_random_data](https://github.com/longhorn/longhorn-tests/blob/a868979cb121ab3c6098ce4fda1c7698e494643a/e2e/libs/volume/crd.py#L266) -> [issue_cmd](https://github.com/longhorn/longhorn-tests/blob/a868979cb121ab3c6098ce4fda1c7698e494643a/e2e/libs/node_exec/node_exec.py#L64C9-L64C18).

Reply:
Triggered https://ci.longhorn.io/job/private/job/longhorn-e2e-test/411/ ~ https://ci.longhorn.io/job/private/job/longhorn-e2e-test/413/

If it somehow becomes flaky, we can still merge this PR to move things forward. The potential of breaking working test cases can be fixed later.

Reply:
I am not sure how often we will hit this, but it should be fixed in https://github.com/longhorn/longhorn-tests/pull/1604/commits/2c269b558ed15f68dafc0ed732b1517fb05c6609. PTAL.